tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -672,6 +692,45 @@ public function jsonRespondedShouldNotContain(
 		Assert::assertEmpty($this->getSpaceByNameFromResponse($spaceName), "space $spaceName should not be available for a user");
 	}
 
+	/**
+	 * @Then /^the json responded should (not|only|)\s?contain spaces of type "([^"]*)"$/
+	 *
+	 * @param string $shoulOrNot (not|only|)
+	 * @param string $type
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function jsonRespondedShouldNotContainSpaceType(
+		string $onlyOrNot,
+		string $type
+	): void {
+		Assert::assertNotEmpty(
+			$spaces = json_decode(
+				(string)$this->featureContext
+					->getResponse()->getBody(),
+				true,
+				512,
+				JSON_THROW_ON_ERROR
+			)
+		);
+		$matches = [];
+		foreach($spaces["value"] as $space) {
+			if($onlyOrNot === "not") {
+				Assert::assertNotEquals($space["driveType"], $type);
+			}
+			if($onlyOrNot === "only") {
+				Assert::assertEquals($space["driveType"], $type);
+			}
```

Comment:
Here confusing `only`. we check that the `DriveType=value` exists. but don't check that there are no other types
 
Maybe use like `@Then /^the json responded should (not|)\s?contain spaces of type "([^"]*)"$/`

Reply:
this is correct. We do this in a foreach loop. In the case of other driveTypes the assertEquals would fail.

So this is correct, you can cross check this.